spring-webflux/src/test/java/org/springframework/web/reactive/function/client/WebClientIntegrationTests.java

```diff
@@ -187,6 +187,64 @@ void retrieveJson(ClientHttpConnector connector) {
 		});
 	}
 
+	@ParameterizedWebClientTest
+	void applyAttributesInNativeRequest(ClientHttpConnector connector) {
+		startServer(connector);
+		connector.setApplyAttributes(true);
+		checkAttributesInNativeRequest(true);
+	}
+
+	@ParameterizedWebClientTest
+	void dontApplyAttributesInNativeRequest(ClientHttpConnector connector) {
+		startServer(connector);
+		connector.setApplyAttributes(false);
+		checkAttributesInNativeRequest(false);
+	}
+
+	private void checkAttributesInNativeRequest(boolean expectAttributesApplied){
+		prepareResponse(response -> {});
+
+		final AtomicReference<Object> nativeRequest = new AtomicReference<>();
+		Mono<Void> result = this.webClient.get()
+				.uri("/pojo")
+				.attribute("foo","bar")
+				.httpRequest(clientHttpRequest -> {
+					nativeRequest.set(clientHttpRequest.getNativeRequest());
+				})
+				.retrieve()
+				.bodyToMono(Void.class);
+		StepVerifier.create(result)
+				.expectComplete()
+				.verify();
+		if (nativeRequest.get() instanceof ChannelOperations<?,?> nativeReq) {
+			Attribute<Map<String, Object>> attributes = nativeReq.channel().attr(AttributeKey.valueOf("attributes"));
+			if (expectAttributesApplied) {
+				assertThat(attributes.get()).isNotNull();
+				assertThat(attributes.get()).containsEntry("foo", "bar");
+			} else {
+				assertThat(attributes.get()).isNull();
+			}
+		} else if (nativeRequest.get() instanceof reactor.netty5.channel.ChannelOperations<?,?> nativeReq) {
+			io.netty5.util.Attribute<Map<String, Object>> attributes = nativeReq.channel().attr(io.netty5.util.AttributeKey.valueOf("attributes"));
+			if (expectAttributesApplied) {
+				assertThat(attributes.get()).isNotNull();
+				assertThat(attributes.get()).containsEntry("foo", "bar");
+			} else {
+				assertThat(attributes.get()).isNull();
+			}
+		} else if (nativeRequest.get() instanceof Request nativeReq) {
+			if (expectAttributesApplied) {
+				assertThat(nativeReq.getAttributes()).containsEntry("foo", "bar");
+			} else {
+				assertThat(nativeReq.getAttributes()).doesNotContainEntry("foo", "bar");
+			}
+		} else if (nativeRequest.get() instanceof org.apache.hc.core5.http.HttpRequest nativeReq) {
+			// TODO get attributes from HttpClientContext
```

Comment:
I could not find out how to verify the request attributes were added to the `org.apache.hc.client5.http.protocol.HttpClientContext` in the Test here, can somebody help out with this?